config/config.go

```diff
@@ -1121,6 +1129,35 @@ func (c *EC2SDConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	return nil
 }
 
+// OpenstackSDConfig is the configuration for OpenStack based service discovery.
+type OpenstackSDConfig struct {
+	IdentityEndpoint string         `yaml:"identity_endpoint"`
+	Username         string         `yaml:"username"`
+	UserID           string         `yaml:"userid"`
+	Password         string         `yaml:"password"`
```

Comment:
Ah, if you rebase this PR against the latest master, you should see a `Secret` type in the `config` package. That was introduced a couple of days ago (fields of type `Secret` are hidden when we show the config in the UI).